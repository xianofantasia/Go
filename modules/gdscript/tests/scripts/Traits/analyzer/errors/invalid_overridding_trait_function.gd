trait SomeTrait:
	func some_func(param: Node):
		print("some func")

class SomeClass:
	uses SomeTrait

	func some_func(param: Node2D):
		print("overridden some func")

func test():
	SomeClass.new().some_func(Node.new())
	print("ok")
