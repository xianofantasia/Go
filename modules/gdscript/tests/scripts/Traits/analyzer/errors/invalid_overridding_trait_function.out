GDTEST_ANALYZER_ERROR
>> ERROR at line 8: "some_func" function signature should be "some_func(Node) -> Variant" as declared in trait "SomeTrait".
